src/tests/ftest/util/dmg_utils.py

```diff
@@ -23,44 +23,183 @@
 """
 from __future__ import print_function
 
-from command_utils import ExecutableCommand
+import os
+import getpass
+
+from command_utils import DaosCommand, CommandWithParameters, CommandFailure
 from command_utils import BasicParameter, FormattedParameter
+from general_utils import get_file_path
+from avocado.utils import process
 
 
-class DmgCommand(ExecutableCommand):
+class DmgCommand(DaosCommand):
     """Defines a object representing a dmg (or daos_shell) command."""
 
     def __init__(self, path):
         """Create a dmg Command object."""
         super(DmgCommand, self).__init__("/run/dmg/*", "daos_shell", path)
 
-        self.request = BasicParameter("{}")
-        self.action = BasicParameter("{}")
-
-        # daos_shell options
         self.hostlist = FormattedParameter("-l {}")
         self.hostfile = FormattedParameter("-f {}")
         self.configpath = FormattedParameter("-o {}")
-        self.insecure = FormattedParameter("-i", None)
-
-        # dmg options
-        self.gid = FormattedParameter("--gid={}")
-        self.uid = FormattedParameter("--uid={}")
-        self.group = FormattedParameter("--group={}")
-        self.mode = FormattedParameter("--mode={}")
-        self.size = FormattedParameter("--size={}")
-        self.nvme = FormattedParameter("--nvme={}")
-        self.svcn = FormattedParameter("--svcn={}")
-        self.target = FormattedParameter("--target={}")
-        self.force = FormattedParameter("--force", False)
-        self.pool = FormattedParameter("--pool={}")
-        self.svc = FormattedParameter("--svc={}")
-        self.rank = FormattedParameter("--rank={}")
-        self.cont = FormattedParameter("--cont={}")
-        self.oid = FormattedParameter("--oid={}")
-
-    def get_param_names(self):
-        """Get a sorted list of dmg command parameter names."""
-        names = self.get_attribute_names(FormattedParameter)
-        names.extend(["request", "action"])
-        return names
+        self.insecure = FormattedParameter("-i", True)
+        self.debug = FormattedParameter("-d", False)
+        self.json = FormattedParameter("-j", False)
+
+    def _get_action_command(self):
+        """Assign a command object for the specified request and action."""
+        if self.action.value == "format":
+            self.action_command = self.DmgFormatSubCommand()
```

Comment:
Don't forget to get the yaml parameters for the prepare sub-command:
```suggestion
            self.action_command = self.DmgFormatSubCommand()
            self.action_command.get_params()
```

Reply:
will change